src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp

```diff
@@ -335,12 +338,35 @@ struct scaled_dot_product_attention_impl : multi_stage_primitive<scaled_dot_prod
             data_inputs_num++;
         }
 
-        params.conf = get_sdpa_configuration(impl_param);
+        auto extend_order = [](const std::vector<int64_t>& order, size_t rank = 4) {
+            if (order.size() >= rank) {
+                return order;
+            }
+            auto diff = rank - order.size();
+            std::vector<int64_t> extended_order(rank, 0);
+            extended_order[diff] = 1;
+
+            for (size_t i = 0, j = 0; i < rank; ++i) {
+                if (i == diff) continue;
+                extended_order[i] = (order[j] == 0) ? 0 : order[j] + 1;
+                ++j;
+            }
```

Comment:
I think we don't need to try to generalize canonicalization here because from the Op spec, only 3D and 4D is supported and when it is 3D, the missing dim is head_num. So just identify head_num dim and set it 1 is enough for 3D case.. (as a simpler way)

Reply:
Updated with a bit of minimization. However, unlike shape, for order, in addition to inserting 1 to heads_num dim, other order values also need to change. e.g. order={1, 0, 2} => {2, 1, 0, 3}  so this is minimal logic.